app/components/channel_drawer/channel_drawer.js

```diff
@@ -192,42 +180,42 @@ export default class ChannelDrawer extends PureComponent {
             viewChannel
         } = actions;
 
-        markChannelAsRead(currentChannel.id);
+        markChannelAsRead(currentChannelId);
```

Comment:
why this action doesn't use the `same` order as selecting channels?

Reply:
This action is more complicated, and it always requires that we contact the server, so I didn't want to change too much and risk breaking anything. I'll see if I can improve it slightly